### PR TITLE
Move validation up before pre-caching/backup

### PR DESCRIPTION
**Pull request by jc-rh**
Also added cgu patch commands to speed up integration test.

### CODE REVIEW

File: controllers/clustergroupupgrade_controller.go [hunk 1]
```diff
@@ -247,296 +310,227 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
...
-					if err != nil {
-						return
-					}
-				}
 			}
 		}
-
 	}
 
 	// Update status
```
Comment (sabbir-47) [inline controllers/clustergroupupgrade_controller.go:536]: We need to handle the error here.

Comment (jc-rh) [reply controllers/clustergroupupgrade_controller.go:536]: Error from where?

Comment (sabbir-47) [reply controllers/clustergroupupgrade_controller.go:536]: Sorry, the comment is not related to this PR, so i couldn't add the comment in line 543. It is in line 543:

	err = r.updateStatus(ctx, clusterGroupUpgrade)
	return

Comment (jc-rh) [reply controllers/clustergroupupgrade_controller.go:536]: Ok. The err is returned. It's not an error we should be handling.

Comment (sabbir-47) [reply controllers/clustergroupupgrade_controller.go:536]: you are right.

**Pull request merged**